src/mpc-hc/PlayerPlaylistBar.h

```diff
@@ -63,6 +63,10 @@ class CPlayerPlaylistBar : public CMPCThemePlayerBar, public CDropClient
     int m_itemHeight = 0;
     int m_initialWindowDPI = 0;
     bool createdWindow;
+    CStringW m_ExternalPlayListPath;
+    CPlaylistIDs m_ExternalPlayListFNCopy;

```

Comment:
With MPCPL, yes it can be saved.  I can't remember if it would reload in that case or not (probably not).

I started to work on code to detect changes to the playlist, but there are a whole bunch of ways it can be modified.  I thought it was safer to validate it was the same rather than hope every possible change to the PL is being tracked.

Reply:
> Comparing a vector of potentially thousands of entries is not really optimal. Maybe instead just store entry count.

I considered that, but if one item were removed and one added, it would make a mistake.  I think the compare is probably pretty fast, but I can benchmark it.

Reply:
Just reading this as a complete outsider... could you not easily compare a hash of the playlist file to a concluding string of the file (e.g. "==== checksum ====" these files are fairly small right?) My apologies for interjecting if I'm just missing something obvious, but have a hash of the file, and if they do in the odd occasion differ, then do whatever comparisons necessary. This way, the comparisons code isn't running each time?

Reply:
> Just reading this as a complete outsider... could you not easily compare a hash of the playlist file to a concluding string of the file (e.g. "==== checksum ====" these files are fairly small right?) My apologies for interjecting if I'm just missing something obvious, but have a hash of the file, and if they do in the odd occasion differ, then do whatever comparisons necessary. This way, the comparisons code isn't running each time?

When MPCHC loads a playlist (m3u), it just copies all the entries into the mpc playlist.  If the user changes it after that, then the current sequence is not really relevant.  But if the playlist stays static, then you could infer that the current position is useful to remember.

For mpcpl files it works differently.  MPC can save them, but I am not sure quite how it works.  I'll test it more later.

Reply:
I benchmarked vector<uint> compares.  If the vectors are identical, it has to compare all entries, so that's the slowest.  For a vector with 999999 entries, the compare took 0.5ms per compare (based on average over 100k attempts).